src/shared/services/label-color.services.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class LabelColorServices {
+    private readonly fixedColors: string[];
+    index!: number;
+    labelColorList!: Map<string, string>;
+
+    constructor() {
```

Comment:
Looks like it is limited to only 20 colors. What happen after that? Is there any other way to get color. maybe using color picker library

Reply:
@devenyantis Hi Deven, the first 20 colors of the labels stay similar to all the projects, 21st color onwards will be randomly generated using the function hue saturation and lightness below there. Previously I did using only randomly pick colors, I find that every time reenters the project, all label colors will randomly pick again, so I am afraid users may confused. That's why I coded this way.

Reply:
I think it doesn't matter. as long as there are color differences. But, anyway, no harm to go as you build as well.
Thanks

Reply:
Thanks @devenyantis